hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/ha/GrpcOMFailoverProxyProvider.java

```diff
@@ -85,14 +87,16 @@ protected void loadOMClientConfigs(ConfigurationSource config, String omSvcId)
         nodeId = OzoneConsts.OM_DEFAULT_NODE_ID;
       }
       if (hostaddr.isPresent()) {
+        int port = hostport.orElse(config
+            .getObject(GrpcOmTransport
+                .GrpcOmTransportConfig.class)
+            .getPort());
         ProxyInfo<T> proxyInfo =
             new ProxyInfo<>(createOMProxy(),
-                hostaddr.get() + ":"
-                    + hostport.orElse(config
-                    .getObject(GrpcOmTransport
-                        .GrpcOmTransportConfig.class)
-                    .getPort()));
+                hostaddr.get() + ":" + port);
         omProxies.put(nodeId, proxyInfo);
+        omNodeAddressMap.put(nodeId, NetUtils.createSocketAddr(hostaddr.get() +
+            ":" + port));
```

Comment:
```
NetUtils.createSocketAddr(hostaddr.get() +
            ":" + port));
```
can be rewritten using the `proxyInfo` we defined above such as,
`NetUtils.createSocketAddr(proxyInfo.proxyInfo));
`